traits/ctraits.pyx

```diff
@@ -1370,15 +1370,8 @@ cdef object getattr_delegate(cTrait trait, CHasTraits obj, object name):
         delegate_attr_name = trait.delegate_attr_name(trait, obj, name)
         tp = type(delegate)
 
-        try:
-            result = getattr(delegate, delegate_attr_name)
-            return result
-        except NotImplementedError as exc:
-            raise exc
-        except Exception as exc:
-            raise DelegationError("The '%.50s' object has no attribute "
```

Comment:
@dpinte I think the other cases never get triggered, but I might be wrong. Looking at the original C code, unless `tp->tp_getattro` is `NULL`, which I think never happens, we always jump over the code that raises the `DelegationError` (via the goto on L1823). On the Cython side, that means we can basically let any exception from the `getattr` call propagate to the caller. Do you know of any cases that would explicitly trigger that `DelegationError`?